cypress/support/commands.js

```diff
@@ -69,6 +92,22 @@ Cypress.Commands.add('mockEventsApi', () => {
     cy.route('**/api/event?status=Live', '@liveEventData').as('getLiveEvents');
 });
 
+Cypress.Commands.add('mockTextFilterResponse', () => {
+    cy.fixture('javaTextFilterResponse.json').as('javaTextFilterResponse');
+    cy.server();
+    cy.route({
```

Comment:
Do you mind explaining what's going on here? Just to confirm my understanding and expand my knowledge :)

Reply:
Absolutely, so for the new text filter on the learn page, what happens is when a user enters a search term we send that search term out to our Realm app which then responds with a set of matching articles for that text filter query.

For testing, we don't exactly want to keep hitting Realm because:
1. Results may change which would break testing
2. Realm may go down which would cause tests to fail

Here, what we do is `stub` the response to this specific endpoint (which corresponds with calling the Realm function for the text filter) with some sample data in `javaTextFilterResponse.json` to populate the articles, letting us not need to rely on hitting the production Realm App every time for results!

Does this make sense?

Reply:
Yep! Thanks!